oauth2client/contrib/gce.py

```diff
@@ -48,28 +45,61 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
+def _get_metadata(http_request=None, path=None):
+    """
     Args:
-        http_request: callable, (Optional) a callable that matches the method
-                      signature of httplib2.Http.request, used to make
-                      the request to the metadata service.
-
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        *path: a list of strings denoting the metadata server request
+            path.
     Returns:
-        tuple, A pair where the first entry is an optional response (from a
-        failed request) and the second is service account email found (as
-        a string).
+        A deserialized JSON object representing the data returned
+        from the metadata server
     """
-    if http_request is None:
+
+    if not path:
+        path = []
+
+    if not http_request:
         http_request = httplib2.Http().request
+
+    full_path = _METADATA_ROOT + '/'.join(path) + '/?recursive=true'
     response, content = http_request(
-        _DEFAULT_EMAIL_METADATA, headers={'Metadata-Flavor': 'Google'})
+        full_path,
+        headers={'Metadata-Flavor': 'Google'}
+    )
     if response.status == http_client.OK:
-        content = _from_bytes(content)
-        return None, content
+        return json.loads(_from_bytes(content))
     else:
-        return response, content
+        msg = (
+            'Failed to retrieve {path} from the Google Compute Engine'
+            'metadata service. Response:\n{error}'
+        ).format(path=full_path, error=response)
+        raise AttributeError(msg)
```

Comment:
I'll just do `ValueError` because that's what's thrown by the `json` module and it's going to be caught and wrapped up as an `HttpAccessTokenError` most of the time anyway